sim/pokemon.ts

```diff
@@ -113,10 +113,10 @@ export class Pokemon {
 
 	/**
 	 * If the switch is called by an effect with a special switch
-	 * message, like U-turn or Baton Pass, this will be the fullname of
+	 * message, like U-turn or Baton Pass, this will be the ID of
 	 * the calling effect.
 	 */
-	switchFlag: boolean | string;
+	switchFlag: ID | boolean;
```

Comment:
Is `switchFlag` ever `true`? If not, this should be `ID | null`, which renders your cast above a significantly easier `!` rather than `as ID`

Reply:
https://github.com/Zarel/Pokemon-Showdown/blob/41d16984527562dc72bbcc46cb6ce85bd1ee2da4/data/abilities.js#L857-L858

Reply:
Very sad. The comment should probably be updated.

Reply:
Hm, is there any harm in setting `switchFlag` to `'emergencyexit' as ID`?

Reply:
There are multiple places where `switchFlag` is set to `true`, not just `Emergency Exit` (`Battle#checkFainted`, `Wimp Out`, `Eject Button` and Gen 4 `Healing Wish` / `Lunar Dance`). Refactoring these to use `ID` instead of `true` is out of scope for this change, IMO

Reply:
Fair.